src/distilabel/steps/tasks/argilla_labeller.py

```diff
@@ -0,0 +1,614 @@
+                "text": "Provide a text response to the question.",
+                "rating": "Provide a rating for the question.",
+            },
+        )
+        labeller.load()
+        ```
+    """
+
+    system_prompt: str = (
+        "You are an expert annotator and labelling assistant that understands complex domains and natural language processing. "
+        "You are given input fields and a question. "
+        "You should create a valid JSON object as an answer to the question based on the input fields. "
+        "1. Understand the input fields and optional guidelines. "
+        "2. Understand the question type and the question settings. "
+        "3. Reason through your response step-by-step. "
+        "4. Provide a valid JSON object as an answer to the question."
+    )
+    question_to_label_instruction: Dict[str, str] = {
+        _LABELQUESTION_SETTINGS.type: "Select the appropriate label from the list of provided labels.",
+        _MULTILABELQUESTION_SETTINGS.type: "Select none, one or multiple labels from the list of provided labels.",
+        _TEXTQUESTION_SETTINGS.type: "Provide a text response to the question.",
+        _RATINGQUESTION_SETTINGS.type: "Provide a rating for the question.",
+    }
+    example_records: Optional[
+        RuntimeParameter[Union[List[Union[Dict[str, Any], Record]], None]]
+    ] = Field(
+        default=None,
+        description="The few shot example records with responses to be used to answer the question.",
+    )
+    fields: Optional[
+        RuntimeParameter[Union[List[Union[TextField, Dict[str, Any]]], None]]
+    ] = Field(
+        default=None,
+        description="The field settings for the fields to be used to answer the question.",
+    )
+    question: Optional[
+        RuntimeParameter[
+            Union[
+                Dict[str, Any],
+                LabelQuestion,
+                MultiLabelQuestion,
+                RatingQuestion,
+                TextQuestion,
+                None,
+            ]
+        ]
+    ] = Field(
+        default=None,
+        description="The question settings for the question to be answered.",
+    )
+    guidelines: Optional[RuntimeParameter[str]] = Field(
+        default=None,
+        description="The guidelines for the annotation task.",
+    )
+
+    _template: Union[Template, None] = PrivateAttr(...)
+    _client: Optional[Any] = PrivateAttr(None)
+
+    def load(self) -> None:
+        """Loads the Jinja2 template."""
+        super().load()
+
+        _path = str(
+            importlib_resources.files("distilabel")
+            / "steps"
+            / "tasks"
+            / "templates"
+            / "argillalabeller.jinja2"
+        )
+
+        self._template = Template(open(_path).read())
+
+    @property
+    def inputs(self) -> List[str]:
+        return ["record"]
+
+    @property
+    def optional_inputs(self) -> List[str]:
+        return ["fields", "question", "example_records", "guidelines"]
```

Comment:
We recently updated the function signature of `inputs` property. Now it returns a `distilabel.steps.typing.StepColumns` which can be either a `List[str]` of mandatory inputs or a `Dict[str, bool]` in which the keys are the names of the inputs and the values are whether if they are mandatory or not. The second case should be perfect for this case.